rs/xnet/payload_builder/src/certified_slice_pool.rs

```diff
@@ -1158,6 +1162,7 @@ impl CertifiedSlicePool {
             Some(entry) => entry,
             None => return Ok(None),
         };
+        let msg_limit = get_msg_limit(subnet_id, state, begin, slice.payload.header.begin());
```

Comment:
If you had a separate function from `get_msg_limit()` (or just an inline calculation) you wouldn't need to pass down `state` all the way here. The 50k limit has nothing to do with the state, it is strictly determined by the incoming stream and `begin`.